feature/topic/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/topic/TopicScreen.kt

```diff
@@ -72,20 +72,24 @@ import com.google.samples.apps.nowinandroid.feature.topic.R.string
 
 @Composable
 fun TopicScreen(
+    topicId: String,
     showBackButton: Boolean,
     onBackClick: () -> Unit,
     onTopicClick: (String) -> Unit,
     modifier: Modifier = Modifier,
-    viewModel: TopicViewModel = hiltViewModel(),
+    viewModel: TopicViewModel = hiltViewModel()
 ) {
+    viewModel.updateTopic(topicId)
```

Comment:
This line feels the strangest with this usage, since we are reusing a single instance of the `ViewModel` scoped to the overall navigation entry with many different `topicId`s that we are swapping between.

One side-effect is that we can't display two different versions of the `TopicScreen` at the same time, which might be ideal to animate between two different selected topics.